fe/fe-core/src/main/java/org/apache/doris/task/ExportExportingTask.java

```diff
@@ -281,67 +199,78 @@ private void registerProfile() {
+            job.cancel(ExportFailMsg.CancelType.RUN_FAIL, "database does not exist");
+            return;
         }
-        TNetworkAddress address = new TNetworkAddress(broker.ip, broker.port);
-        TPaloBrokerService.Client client = null;
-        try {
-            client = ClientPool.brokerPool.borrowObject(address);
-        } catch (Exception e) {
-            try {
-                client = ClientPool.brokerPool.borrowObject(address);
-            } catch (Exception e1) {
-                String failMsg = "create connection to broker(" + address + ") failed";
-                LOG.warn(failMsg);
-                return new Status(TStatusCode.CANCELLED, failMsg);
-            }
+
+        if (job.isReplayed()) {
+            // If the job is created from replay thread, all plan info will be lost.
+            // so the job has to be cancelled.
+            String failMsg = "FE restarted or Master changed during exporting. Job must be cancelled.";
+            job.cancel(ExportFailMsg.CancelType.RUN_FAIL, failMsg);
+            return;
+        }
+
+        // make snapshots
+        Status snapshotStatus = makeSnapshots();
+        if (!snapshotStatus.ok()) {
+            job.cancel(ExportFailMsg.CancelType.RUN_FAIL, snapshotStatus.getErrorMsg());
+            return;
+        }
+
+        if (job.updateState(ExportJob.JobState.EXPORTING)) {
+            LOG.info("Exchange pending status to exporting status success. job: {}", job);
+            return;
         }
-        boolean failed = false;
-        Set<String> exportedFiles = job.getExportedFiles();
-        List<String> newFiles = Lists.newArrayList();
-        String exportPath = job.getExportPath();
-        for (String exportedFile : exportedFiles) {
-            // move exportPath/__doris_tmp/file to exportPath/file
-            String file = exportedFile.substring(exportedFile.lastIndexOf("/") + 1);
-            String destPath = exportPath + "/" + file;
-            LOG.debug("rename {} to {}, export job: {}", exportedFile, destPath, job.getId());
-            String failMsg = "";
-            try {
-                TBrokerRenamePathRequest request = new TBrokerRenamePathRequest(
-                        TBrokerVersion.VERSION_ONE, exportedFile, destPath, job.getBrokerDesc().getProperties());
-                TBrokerOperationStatus tBrokerOperationStatus = null;
-                tBrokerOperationStatus = client.renamePath(request);
-                if (tBrokerOperationStatus.getStatusCode() != TBrokerOperationStatusCode.OK) {
-                    failed = true;
-                    failMsg = "Broker renamePath failed. srcPath=" + exportedFile + ", destPath=" + destPath
-                            + ", broker=" + address  + ", msg=" + tBrokerOperationStatus.getMessage();
-                    return new Status(TStatusCode.CANCELLED, failMsg);
-                } else {
-                    newFiles.add(destPath);
+    }
+
+    private Status makeSnapshots() {
```

Comment:
Currently, we can comment out this `makeSnapshots()` method because it is not used.
It can be reopened after we implement exporting tablet one by one

Reply:
done